dockstore-webservice/src/main/java/io/dockstore/webservice/core/AppTool.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *    Copyright 2019 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.dockstore.webservice.core;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+import io.dockstore.common.EntryType;
+import io.swagger.annotations.ApiModel;
+
+@ApiModel(value = "AppTool", description = "This describes one service in the dockstore as a special degenerate case of a workflow", parent = Workflow.class)
+@Entity
```

Comment:
This might be the reason for the migration/constraint issues. There are annotations for constraints on tables
For example https://github.com/dockstore/dockstore/blob/d34af609a98da26d6af2fdf57ae6f6d4362a96df/dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowVersion.java#L50

Reply:
not this